src/iceberg/manifest_list.h

```diff
@@ -0,0 +1,202 @@
+  /// Number of entries in the manifest that have status ADDED (1), when null this is
+  /// assumed to be non-zero
+  std::optional<int32_t> added_files_count;
+  /// Field id: 505
+  /// Number of entries in the manifest that have status EXISTING (0), when null this is
+  /// assumed to be non-zero
+  std::optional<int32_t> existing_files_count;
+  /// Field id: 506
+  /// Number of entries in the manifest that have status DELETED (2), when null this is
+  /// assumed to be non-zero
+  std::optional<int32_t> deleted_files_count;
+  /// Field id: 512
+  /// Number of rows in all of files in the manifest that have status ADDED, when null
+  /// this is assumed to be non-zero
+  std::optional<int64_t> added_rows_count;
+  /// Field id: 513
+  /// Number of rows in all of files in the manifest that have status EXISTING, when null
+  /// this is assumed to be non-zero
+  std::optional<int64_t> existing_rows_count;
+  /// Field id: 514
+  /// Number of rows in all of files in the manifest that have status DELETED, when null
+  /// this is assumed to be non-zero
+  std::optional<int64_t> deleted_rows_count;
+  /// Field id: 507
+  /// Element field id: 508
+  /// A list of field summaries for each partition field in the spec. Each field in the
+  /// list corresponds to a field in the manifest file's partition spec.
+  std::vector<FieldSummary> partitions;
+  /// Field id: 519
+  /// Implementation-specific key metadata for encryption
+  std::vector<uint8_t> key_metadata;
+  /// Field id: 520
+  /// The starting _row_id to assign to rows added by ADDED data files
+  int64_t first_row_id;
+
+  /// \brief Checks if this manifest file contains entries with ADDED status.
+  [[nodiscard]] bool has_added_files() const {
+    return added_files_count.has_value() && *added_files_count > 0;
+  }
+
+  /// \brief Checks if this manifest file contains entries with EXISTING status.
+  [[nodiscard]] bool has_existing_files() const {
+    return existing_files_count.has_value() && *existing_files_count > 0;
+  }
+
+  /// \brief Checks if this manifest file contains entries with DELETED status
+  [[nodiscard]] bool has_deleted_files() const {
+    return deleted_files_count.has_value() && *deleted_files_count > 0;
+  }
+
+  static const SchemaField MANIFEST_PATH;
+  static const SchemaField MANIFEST_LENGTH;
+  static const SchemaField PARTITION_SPEC_ID;
+  static const SchemaField CONTENT;
+  static const SchemaField SEQUENCE_NUMBER;
+  static const SchemaField MIN_SEQUENCE_NUMBER;
+  static const SchemaField ADDED_SNAPSHOT_ID;
+  static const SchemaField ADDED_FILES_COUNT;
+  static const SchemaField EXISTING_FILES_COUNT;
+  static const SchemaField DELETED_FILES_COUNT;
+  static const SchemaField ADDED_ROWS_COUNT;
+  static const SchemaField EXISTING_ROWS_COUNT;
+  static const SchemaField DELETED_ROWS_COUNT;
+  static const SchemaField PARTITIONS;
+  static const SchemaField KEY_METADATA;
+  static const SchemaField FIRST_ROW_ID;
+
+  static StructType Schema();
```

Comment:
```suggestion
  static const StructType& Type();
```

Same as the above comment

Reply:
done.